src/libraries/System.Data.OleDb/src/ColumnBinding.cs

```diff
@@ -631,7 +631,9 @@ private string Value_BSTR()
             string value = "";
             RowBinding bindings = RowBinding;
             bool mustRelease = false;
+#if !NETCOREAPP
```

Comment:
This is done to avoid an obsoletion error?

I wonder if on .NET Core we should not only obsolete this API, but also make it Conditonal("SOMETHING NO ONE WOULD EVER WRITE"), so that it gets compiled away but still compiles.  @jkotas, opinion?

Reply:
JIT will trivially inline the empty call. It is miniscule IL size optimization. Personally, I would not spend any cycles on improving these obsolete APIs (CERs, CAS, ...). If somebody wants to spend time on it, I am fine with it.

Reply:
I removed all the ifdefs except the ones which use Span/Memory in combination with string.Concat for perf and the ones around `RuntimeHelpers.PrepareConstrainedRegions();`. The latter as that's consistent with what we did for CAS invocation in other libs like System.Reflection.Metadata: https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/PinnedObject.cs#L20-L26.

Reply:
I don't know why we did that in S.R.Metadata, but I still think it'd be better in this PR to not add all of these ifdefs, and instead just suppress SYSLIB0004 in the .csproj.

Reply:
> I don't know why we did that in S.R.Metadata

Sure, again I don't have a strong opinion on the change. Should we then revert what was done in S.R.Metdata for consistency?